apps/vmq_enhanced_auth/src/vmq_enhanced_auth.erl

```diff
@@ -954,4 +966,139 @@ simple_acl(_) ->
+            ]},
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {Topic1, 0}
+                ]
+            )
+        ),
+        ?_assertEqual(
+            {ok, [
+                {Topic2, 0, {matched_acl, undefined, <<"abc/xyz/#">>}}
+            ]},
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {Topic2, 0}
+                ]
+            )
+        )
+    ].
+delete_complex_acl_test(_) ->
+    ACL = [<<"topic abc/xyz/#\n">>],
+    load_from_list(ACL),
+    Topic1 = [<<"abc">>, <<"xyz">>, <<"+">>, <<"1">>],
+    Topic2 = [<<"abc">>, <<"xyz">>, <<"+">>, <<"2">>],
+    SubTopic = [<<"abc">>, <<"xyz">>, <<"+">>],
+    vmq_reg_redis_trie:add_complex_topic("", Topic1),
+    vmq_reg_redis_trie:add_complex_topic("", Topic2),
+    vmq_reg_redis_trie:delete_complex_topic("", Topic1),
+    vmq_reg_redis_trie:delete_complex_topic("", SubTopic),
+    [
+        ?_assertEqual(
+            next,
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {Topic1, 0}
+                ]
+            )
+        ),
+        ?_assertEqual(
+            {ok, [
+                {Topic2, 0, {matched_acl, undefined, <<"abc/xyz/#">>}}
+            ]},
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {Topic2, 0}
+                ]
+            )
+        )
+    ].
+subtopic_subscribe_test(_) ->
+    ACL = [<<"topic abc/xyz/#\n">>],
+    load_from_list(ACL),
+    Topic = [<<"abc">>, <<"xyz">>, <<"+">>, <<"1">>, <<"+">>],
+    SubTopic1 = [<<"abc">>, <<"xyz">>, <<"+">>, <<"1">>],
+    SubTopic2 = [<<"abc">>, <<"xyz">>, <<"+">>],
+    vmq_reg_redis_trie:add_complex_topic("", Topic),
+    vmq_reg_redis_trie:add_complex_topic("", SubTopic1),
+    [
+        ?_assertEqual(
+            {ok, [
+                {Topic, 0, {matched_acl, undefined, <<"abc/xyz/#">>}}
+            ]},
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {Topic, 0}
+                ]
+            )
+        ),
+        ?_assertEqual(
+            {ok, [
+                {SubTopic1, 0, {matched_acl, undefined, <<"abc/xyz/#">>}}
+            ]},
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {SubTopic1, 0}
+                ]
+            )
+        ),
+        ?_assertEqual(
+            next,
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {SubTopic2, 0}
+                ]
+            )
+        )
+    ].
+get_complex_topics_test(_) ->
```

Comment:
These tests are for complex trie which is not implemented in vmq_enhanced_auth plugin.

Reply:
Removed